tests/queryset_pickle/tests.py

```diff
@@ -219,6 +219,40 @@ def test_pickle_subquery_queryset_not_evaluated(self):
         with self.assertNumQueries(0):
             self.assert_pickles(groups)
 
+    def test_pickle_filteredrelation(self):
+        group = Group.objects.create(name='group')
+        event_1 = Event.objects.create(title='Big event', group=group)
+        event_2 = Event.objects.create(title='Small event', group=group)
+        Happening.objects.bulk_create([
+            Happening(event=event_1, number1=5),
+            Happening(event=event_2, number1=3),
+        ])
+        groups = Group.objects.annotate(
+            big_events=models.FilteredRelation(
+                'event',
+                condition=models.Q(event__title__startswith='Big'),
+            ),
+        ).annotate(sum_number=models.Sum('big_events__happening__number1'))
+        groups_query = pickle.loads(pickle.dumps(groups.query))
+        groups = Group.objects.all()
+        groups.query = groups_query
+        self.assertEqual(groups.get().sum_number, 5)
+
+    def test_pickle_filteredrelation_m2m(self):
+        group = Group.objects.create(name='group')
+        m2mmodel = M2MModel.objects.create()
+        m2mmodel.groups.add(group)
+        groups = Group.objects.annotate(
+            first_m2mmodels=models.FilteredRelation(
+                'm2mmodel',
+                condition=models.Q(m2mmodel__pk__lt=10),
```

Comment:
This condition is problematic on databases that use random rather than serial pk values (failure observed on CockroachDB).

Reply:
@timgraham if the `M2MModel` created above had a fixed `pk` would that resolve your issue?

Reply:
Yes

Reply:
Thanks for the report, I've prepare a fix #13529.